vmdb/app/controllers/api_controller.rb

```diff
@@ -4,6 +4,17 @@ class Forbidden                 < StandardError; end
   class BadRequestError           < StandardError; end
   class UnsupportedMediaTypeError < StandardError; end
 
+  def handle_options_request
+    head(:ok) if request.request_method == "OPTIONS"
+  end
+
+  after_filter :set_access_control_headers
+  def set_access_control_headers
+    headers['Access-Control-Allow-Origin'] = '*'  # FIXME?
```

Comment:
According to docs, "The value of "*" is special in that it does not allow requests to supply credentials, meaning HTTP authentication, client-side SSL certificates, nor does it allow cookies to be sent." Might need to specify the URL here. Let's test it with Basic Auth, as well as authentication token (X-Auth-Token).